src/main/java/com/scottlogic/sorts/AuthorPostSorter.java

```diff
@@ -16,17 +18,22 @@
     public List<UserPost> sort(List<UserPost> inputList, SortOrder orderIn) {
 
         List<UserPost> listToSort = new ArrayList<>();
+        List<UserPost> listSorted = new ArrayList<>();
+
         if (inputList == null) {
             return listToSort;
         }
         listToSort = NullPostChecker.nullPostCheck(inputList);
 
+
         switch (orderIn) {
 
-            case ASC -> Collections.sort(listToSort, comparing(UserPost::getAuthor, nullsLast(naturalOrder())));
+            case ASC -> listSorted = listToSort.stream().sorted(Comparator.nullsLast(Comparator.comparing(UserPost::getAuthor, nullsLast(naturalOrder())))).collect(Collectors.toList());
 
-            case DESC -> Collections.sort(listToSort, Collections.reverseOrder(comparing(UserPost::getAuthor, nullsFirst(naturalOrder()))));
+
+            case DESC -> listSorted = listToSort.stream().sorted(Collections.reverseOrder(comparing(UserPost::getAuthor, nullsFirst(naturalOrder())))).collect(Collectors.toList());
```

Comment:
We could also potentially get rid of the switch statement by adding an `isAscending()` method to the enum.

Reply:
Putting an isAscending() method caused a lot of issues. So i'm still using a ternary operator and done it but without a method

Reply:
I think the switch is preferable to this. We can have a chat about the `isAscending()` method if you like.

Reply:
Yes please